app/buck2_client_ctx/src/subscribers/superconsole.rs

```diff
@@ -1001,10 +1001,20 @@ fn truncate(contents: &str) -> Option<String> {
     const MAX_LENGTH: usize = 200;
     const BUFFER: usize = " ...<omitted>... ".len();
     if contents.len() > MAX_LENGTH + BUFFER {
+        let first_slice_end = contents
+            .char_indices()
+            .map(|(i, _)| i)
+            .nth(MAX_LENGTH / 2)
+            .unwrap();
```

Comment:
Don't think this can be right. `contents.len()` is in bytes, you can have 200 bytes but <100 characters.

I think maybe you just want `.floor_char_boundary()`

Reply:
> Don't think this can be right. contents.len() is in bytes, you can have 200 bytes but <100 characters.

Yeah, this is trading off one potential panic for another, less likely one.

> I think maybe you just want .floor_char_boundary()

I do! Thanks for pointing that out.
I've used ceil for the first part and floor for the second one so that it's less likely that it will split in the middle of a grapheme, but I'm happy to switch that if you prefer.